book/embeds.md

```diff
@@ -333,71 +334,93 @@ class InputLayout(EmbedLayout):
         self.height = linespace(self.font)
 ```
 
-And `ImageLayout` is almost the same. The two key differences
-are the need to actually load the image off the network, and then use
-the size of the image to size the `ImageLayout`. Let's start with loading.
-After loading, the image is stored on the `node`. But this adds some complexity
-in the `image` function we need to add on `BlockLayout`, because first it needs
-to load the image and only then set its `parent` and `previous` fields. That'll
-be via a new `init` method call.
+And `ImageLayout` is almost the same. In `InputLayout`, the new `image`
+method is actually almost a carbon copy of `input` and `text`. So now we have
+three methods that are almost identical for each atomic piece of inline content.
+The two things in common are computing the font  (and passing it to the various
+functions):
 
 ``` {.python}
 class BlockLayout(LayoutObject):
+    def font(self, node, zoom):
+        weight = node.style["font-weight"]
+        style = node.style["font-style"]
+        font_size = device_px(float(node.style["font-size"][:-2]), zoom)
+        return get_font(font_size, weight, font_size)
+
     def recurse(self, node, zoom):
-            # ...
-            elif node.tag == "img":
-                self.image(node, zoom)
-    
-    def image(self, node, zoom):
-        img = ImageLayout(node, self.frame)
-        w = device_px(node.image.width(), zoom)
+        if isinstance(node, Text):
+            self.text(node, zoom)
+        else:
+            if node.tag == "br":
+                self.new_line()
+            elif node.tag == "input" or node.tag == "button":
+                self.input(node, zoom)
+            else:
+                for child in node.children:
+                    self.recurse(child, zoom)
```

Comment:
Now, this part doesn't change, right? We're just adding a `font` method?

Also, IIUC the `font` method doesn't rely on the `BlockLayout` at all. It's fine to keep it here, but what do you think of moving it to a stand-alone method or a method on `Element` or `Text` or something? That can be a different PR.

Reply:
good point. I'll delete the recurse part in this PR at least.